packages/rum-core/src/domain/tracing/tracer.spec.ts

```diff
@@ -132,6 +134,90 @@ describe('tracer', () => {
+        ...INIT_CONFIGURATION,
+        configureTracingUrls: [{ match: window.location.origin, headerTypes: ['b3m'] }],
+      })!
+
+      const tracer = startTracer(configurationWithB3Multi, sessionManager)
+      const context = { ...ALLOWED_DOMAIN_CONTEXT }
+      tracer.traceXhr(context, xhrStub as unknown as XMLHttpRequest)
+
+      expect(xhrStub.headers['X-B3-TraceId']).toBeDefined()
+      expect(xhrStub.headers['X-B3-SpanId']).toBeDefined()
+      expect(xhrStub.headers['X-B3-Sampled']).toBeDefined()
+
+      expect(xhrStub.headers['x-datadog-origin']).toBeUndefined()
+      expect(xhrStub.headers['x-datadog-parent-id']).toBeUndefined()
+      expect(xhrStub.headers['x-datadog-trace-id']).toBeUndefined()
+      expect(xhrStub.headers['x-datadog-sampling-priority']).toBeUndefined()
+    })
+
+    it('should add B3 (single) and W3C headers', () => {
+      const configurationWithB3andW3C = validateAndBuildRumConfiguration({
+        ...INIT_CONFIGURATION,
+        configureTracingUrls: [{ match: window.location.origin, headerTypes: ['b3', 'w3c'] }],
+      })!
+
+      const tracer = startTracer(configurationWithB3andW3C, sessionManager)
+      const context = { ...ALLOWED_DOMAIN_CONTEXT }
+      tracer.traceXhr(context, xhrStub as unknown as XMLHttpRequest)
+
+      expect(xhrStub.headers['b3']).toBeDefined()
+      expect(xhrStub.headers['traceparent']).toBeDefined()
+    })
+
+    it('should not add any headers', () => {
+      const configurationWithoutHeaders = validateAndBuildRumConfiguration({
+        ...INIT_CONFIGURATION,
+        configureTracingUrls: [{ match: window.location.origin, headerTypes: [] }],
+      })!
+
+      const tracer = startTracer(configurationWithoutHeaders, sessionManager)
+      const context = { ...ALLOWED_DOMAIN_CONTEXT }
+      tracer.traceXhr(context, xhrStub as unknown as XMLHttpRequest)
+
+      expect(xhrStub.headers['b3']).toBeUndefined()
+      expect(xhrStub.headers['traceparent']).toBeUndefined()
+      expect(xhrStub.headers['x-datadog-trace-id']).toBeUndefined()
+      expect(xhrStub.headers['X-B3-TraceId']).toBeUndefined()
+    })
+
+    it('should ignore wrong header types', () => {
+      const configurationWithBadParams = validateAndBuildRumConfiguration({
+        ...INIT_CONFIGURATION,
+        configureTracingUrls: [{ match: window.location.origin, headerTypes: ['foo', 32, () => true] as any }],
+      })!
+
+      const tracer = startTracer(configurationWithBadParams, sessionManager)
+      const context = { ...ALLOWED_DOMAIN_CONTEXT }
+      tracer.traceXhr(context, xhrStub as unknown as XMLHttpRequest)
+
+      expect(xhrStub.headers['b3']).toBeUndefined()
+      expect(xhrStub.headers['traceparent']).toBeUndefined()
+      expect(xhrStub.headers['x-datadog-trace-id']).toBeUndefined()
+      expect(xhrStub.headers['X-B3-TraceId']).toBeUndefined()
+    })
+
+    it('should survive a wrong match type', () => {
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏add a test case with a `match` function that throws an error, to test for the `display.error` code path

Reply:
👌